spec/factories/components.rb

```diff
@@ -1,13 +1,16 @@
 FactoryBot.define do
+
   factory :sp_component do
     component_type { COMPONENT_TYPE::SP }
     name { 'Test Service Provider' }
+#    team { create(:team) }
     environment { 'staging' }
   end
 
   factory :msa_component do
     component_type { COMPONENT_TYPE::MSA }
     entity_id { 'https://test-entity-id' }
+#    team { create(:team) }
```

Comment:
was it not possible to do 

`team_id { create(:team).id }`

That would potentially mean we wouldn't need to pass in team_id into all the creates in the specs

Reply:
It doesn't cause any problems, but it also doesn't seem to fix any; even with this change in place, we still need the creates. :(

Reply:
it should be` team_id { 1 } ` or whatever string since we're not validating for the existence of the team at this point... and it should be on the events factories (`factory :new_sp_component_event do ...`)

Reply:
or actually let's do the proper team create `team_id { create(:team).id }` so it's easier once we add the validation later on